### DISCUSSION TITLE
Multithreaded CBC solving using ortools pywraplp

### DISCUSSION THREAD

**Discussion by goodman-dev** [google/or-tools#4344] [Build and configuration issues]
Hello. I hope this is the correct place for this question, apologies if not.

I'm using ortools' pywraplp to solve MILP problems in Python, as below, with `ortools` installed through Pip.

```python
from ortools.linear_solver import pywraplp

solver = pywraplp.Solver.CreateSolver('CBC_MIXED_INTEGER_PROGRAMMING')
solver.SetNumThreads(4)
# ...
# constraints
# ...
status = solver.Solve()
```

I get the following output when running this, seeming to indicate that the underlying CBC solver doesn't have multithread support. I'm on Mac, and have tested both on ARM & x86.

```
No match for threads - ? for list of commands
No match for 4 - ? for list of commands
```

Is there a way to instruct pywrap to use specific Cbc binaries/shared-libraries, or alternatively to compile ortools & pywrap from source with multithreading enabled?

I've compiled Cbc with multithreading support myself and exported my model to LP format, and have confirmed that enabling multithreading makes a significant difference to the performance, so I'm very keen to utilise this capability through Python.

**Comment by lperron** [comment]
it was crashing, leaking memory... We disabled it.
